src/lando/api/legacy/workers/uplift_worker.py

```diff
@@ -0,0 +1,235 @@
+        job_url = job.url()
+
+        try:
+            created_revision_ids = self.apply_and_uplift(job)
+        except TemporaryFailureException:
+            return False
+        except PermanentFailureException as exc:
+            self.notify_uplift_failure(job, repo.name, job_url, user.email, str(exc))
+            return False
+        except Exception as exc:  # pragma: no cover - defensive catch
+            self.notify_uplift_failure(job, repo.name, job_url, user.email, str(exc))
+            return False
+
+        self.notify_success(
+            repo.name,
+            job_url,
+            user.email,
+            created_revision_ids,
+        )
+        return True
+
+    def apply_and_uplift(self, job: UpliftJob) -> list[int]:
+        """Apply uplift patches to the repo and create new revisions.
+
+        Returns an ordered list of created revision IDs.
+        """
+        repo = job.target_repo
+        multi_request = job.multi_request
+        user = multi_request.user
+        scm = repo.scm
+
+        # Update to the latest commit in the target train.
+        base_revision = self.update_repo(repo, job, scm, target_cset=None)
+
+        # Apply patches to the tip of the target train.
+        def apply_patch(revision: Revision):
+            logger.debug(f"Landing {revision} ...")
+            scm.apply_patch(
+                revision.diff,
+                revision.commit_message,
+                revision.author,
+                revision.timestamp,
+            )
+
+        for uplift_revision in job.revisions.all():
+            self.handle_new_commit_failures(
+                apply_patch, repo, job, scm, uplift_revision
+            )
+            new_commit = scm.describe_commit()
+            logger.debug(f"Created new commit {new_commit}")
+
+        # On success: create patches.
+        response = self.moz_phab_uplift(
+            job, user.profile.phabricator_api_key, base_revision
+        )
+
+        # Retrieve created revision IDs and tip revision ID.
+        commits = response["commits"]
+        created_revision_ids = [int(commit["rev_id"]) for commit in commits]
+        tip_revision_id = created_revision_ids[-1]
+
+        _, created = UpliftRevision.objects.get_or_create(
+            revision_id=tip_revision_id,
+            defaults={"assessment": multi_request.assessment},
+        )
+        if not created:
+            UpliftRevision.objects.filter(revision_id=tip_revision_id).update(
+                assessment=multi_request.assessment
+            )
+
+        # Trigger a Celery task to update the form on Phabricator.
+        self.call_task(
+            set_uplift_request_form_on_revision,
+            tip_revision_id,
+            multi_request.assessment.to_conduit_json_str(),
+            user.id,
+        )
+
+        job.created_revision_ids = created_revision_ids
+        job.transition_status(JobAction.SUCCESS)
+
+        return created_revision_ids
+
+    def notify_success(
```

Comment:
nit: for consistency with the failure version of this
```suggestion
    def notify_uplift_success(
```